libsolidity/interface/StandardCompiler.cpp

```diff
@@ -1164,8 +1180,102 @@ std::map<std::string, Json::Value> StandardCompiler::parseAstFromInput(StringMap
 	return sourceJsons;
 }
 
+Json::Value StandardCompiler::importEVMAssembly(StandardCompiler::InputsAndSettings _inputsAndSettings)
+{
+	solAssert(_inputsAndSettings.language == "EVMAssembly");
+	solAssert(_inputsAndSettings.sources.empty());
+	solAssert(_inputsAndSettings.jsonSources.size() == 1);
+
+	if (!isBinaryRequested(_inputsAndSettings.outputSelection))
+		return Json::objectValue;
+
+	evmasm::EVMAssemblyStack stack(_inputsAndSettings.evmVersion);
+	std::string const& sourceName = _inputsAndSettings.jsonSources.begin()->first; // result of structured binding can only be used within lambda from C++20 on.
+	Json::Value const& sourceJson = _inputsAndSettings.jsonSources.begin()->second;
+	try
+	{
+		stack.analyze(sourceName, sourceJson);
+		stack.assemble();
+	}
+	catch (util::Exception const& e)
+	{
+		return formatFatalError(Error::Type::Exception, "Exception during assembly import: " + std::string(e.what()));
+	}
+	if (!stack.compilationSuccessful())
+		return Json::objectValue;
+
+	// EVM
+	bool const wildcardMatchesExperimental = false;
+	Json::Value evmData = Json::objectValue;
+	if (isArtifactRequested(_inputsAndSettings.outputSelection, sourceName, "", "evm.assembly", wildcardMatchesExperimental))
+		// not optimal: we need to build a string map, containing the original source as string.
+		evmData["assembly"] = stack.assemblyString(sourceName, {{sourceName, util::jsonCompactPrint(sourceJson)}});
```

Comment:
Is it even correct? You're passing serialized JSON as source. When compiling Solidity code, this would be the `.sol` files and they'd be there only so that we can include snippets from the original file if requested via `settings.debug.debugInfo` setting. If you shove the JSON in here, the snippets will not make any sense.

Well, when doing assembly import we no longer have access to the original source. You should leave it empty and it should be optional if it isn't already. Also, we should add an unimplemented assert or the `debugInfo` setting. Requesting anything via it won't work so we should really accept only empty values. And we should change it's default value for import.

Reply:
changed to `evmData["assembly"] = stack.assemblyString(sourceName, {{}});`